runtime/vm/NativeHelpers.cpp

```diff
@@ -149,6 +149,17 @@ convertCStringToByteArray(J9VMThread *currentThread, const char *cString)
 	return result;
 }
 
+UDATA *
+convertToNativeArgArray(J9VMThread *currentThread, j9object_t argArray, UDATA *ffiArgs)
+{
+	UDATA argCount = (UDATA)J9INDEXABLEOBJECT_SIZE(currentThread, argArray);
+	/* 3 means each element of the array to be copied is 8 bytes (64bits) in size
```

Comment:
So does this mean the type is not actually `UDATA*`? Or does the value need to change between 32 and 64 bit?

Reply:
We already covert all parameters to `long` at first in `ProgrammableInvoker` do which means it must be 64 bit in there.

Reply:
Then the type should really be `jlong`.

Reply:
You're getting lucky because I bet you've only run this on a little endian platform. The array is being accessed as UDATA which will certainly not work in most cases.

Reply:
> Then the type should really be jlong

I am not quite following this (why it should be `jlong` which is a JNI type). 

> You're getting lucky because I bet you've only run this on a little endian platform. The array is being accessed as UDATA which will certainly not work in most cases.

Actually we have code partially set up on `AIX/PPC` which works good (except the Valist support) without any error.
We already converted the parameter to specific types required in ffi_call as follows:
```
inlProgrammableInvokerInvokeNative
#if !defined(J9VM_ENV_LITTLE_ENDIAN)
				/* Note: A float number is converted to int by Float.floatToIntBits() in ProgrammableInvoker */
				if ((J9NtcInt == argType) || (J9NtcBoolean == argType) || (J9NtcFloat == argType)) {
					values[i] = (void *)((UDATA)values[i] + extraBytesOfInt);
				} else if ((J9NtcShort == argType) || (J9NtcChar == argType)) {
					values[i] = (void *)((UDATA)values[i] + extraBytesOfShortAndChar);
				} else if (J9NtcByte == argType) {
					values[i] = (void *)((UDATA)values[i] + extraBytesOfByte);
				}
```

Reply:
I misspoke - it won't work on 32 bit where UDATA is only 4 bytes.

Reply:
I don't think we need to support 32 bit. 
Am I correct at this point, @tajila?

Reply:
Even if we decide that, you should use the correct type U_64 instead of UDATA.

Reply:
Agreed and replaced with `U_64` in such case.

Reply:
There are currently no plans to support 32bit for JDK11+ at the moment. However, there is a small possibility that we may need to support z/OS-31bit.

Reply:
There is no code changes for now to address this case. So might need to add a little code when it comes up.